src/sentry/api/bases/organization.py

```diff
@@ -389,23 +389,23 @@ def _filter_projects_by_permissions(
                 span.set_tag("mode", "force_global_perms")
                 return projects
 
+            # There is a special case for staff, where we want to fetch a single project (OrganizationStatsEndpointV2)
+            # or all projects (OrganizationMetricsDetailsEndpoint) in _admin. Staff cannot use has_project_access
+            # like superuser because it fails dueto staff having no scopes. The workaround is to create a lambda that
+            # mimics checking for active projects like has_project_access without further validation.
+            # NOTE: We must check staff before superuser or else _admin will fail when both cookies are active
+            if is_active_staff(request):
```

Comment:
I thought this check could be nested inside elif not filter_by_membership: but we actually use it in _admin for fetching all projects.
The check also needs to be placed before superuser or else _admin will fail when both cookies are active.